### PR TITLE
Comp index select

### PR DESCRIPTION
**Pull request by sljlp**
### PR types

Others
### PR changes

Others
### Describe

add composite index_select op

### CODE REVIEW

File: python/paddle/incubate/autograd/primitives.py [hunk 1]
```diff
@@ -124,5 +127,7 @@
     'full',
     'uniform',
     'greater_equal',
+    'stack'
```
Comment (cyber-pioneer) [inline python/paddle/incubate/autograd/primitives.py]: why import these ops? they are not used in this composite rule.

**Review by cyber-pioneer — DISMISSED** [review]
ensure code coverage

File: paddle/phi/kernels/gpu/index_select_grad_kernel.cu [hunk 2]
```diff
@@ -31,7 +31,6 @@ template <typename T, typename IndexT>
 __global__ void index_select_grad_cuda_kernel(const T* output_grad,
                                               T* input_grad,
                                               const IndexT* index,
-                                              int64_t nums,
```
Comment (cyber-pioneer) [inline paddle/phi/kernels/gpu/index_select_grad_kernel.cu]: why to delete this?

File: python/paddle/incubate/autograd/composite_rules.py [hunk 3]
```diff
@@ -300,6 +300,15 @@ def hard_swish_composite(x):
     return res
 
 
+@REGISTER_COMPOSITE('index_select')
+def index_select_composite(x, index, axis=0):
+    """define composite rule of op index_select."""
```
Comment (cyber-pioneer) [inline python/paddle/incubate/autograd/composite_rules.py:394]: composite args will be overwritten. Thus  do not set default here.

File: paddle/phi/kernels/gpu/index_select_grad_kernel.cu (hunk 2 above)
Comment (sljlp) [reply paddle/phi/kernels/gpu/index_select_grad_kernel.cu]: not used in this kernel function.

File: python/paddle/incubate/autograd/composite_rules.py (hunk 3 above)
Comment (sljlp) [reply python/paddle/incubate/autograd/composite_rules.py:394]: updated

**Review by Charles-hit — APPROVED** [review]

**Pull request merged**